src/context.ts

```diff
@@ -53,14 +52,22 @@ export class OperonContextImpl implements OperonContext {
   /*** Application configuration ***/
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   applicationConfig?: any;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getConfig(key: string): any {
-    if (!this.applicationConfig) {
+  getConfig<T>(key: string, defaultValue?: T): T | undefined {
```

Comment:
```suggestion
  getConfig<T>(key: string): T | undefined;
  getConfig<T>(key: string, defaultValue: T): T;
  getConfig<T>(key: string, defaultValue?: T): T | undefined {
```

Reply:
Also add the overload declarations to the impl class so that we can use the overloads inside the implementation if needed